src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -407,14 +427,14 @@ static bool FilterMatches(DeclaredSymbolInfo info, SymbolFilter filter)
             return results.Any(b => b);
         }
 
-        private static readonly Func<DocumentId, Project, Document?> s_tryCreateDocumentFunction =
-            (documentId, project) => project._projectState.DocumentStates.TryGetState(documentId, out var state) ? new Document(project, state) : null;
+        private static readonly Func<DocumentId, Project, Document> s_tryCreateDocumentFunction =
```

Comment:
Applies to all `s_try*` which are no longer conditional

```suggestion
        private static readonly Func<DocumentId, Project, Document> s_createDocumentFunction =
```